src/frontend/handlers.go

```diff
@@ -72,6 +72,20 @@ func (fe *frontendServer) homeHandler(w http.ResponseWriter, r *http.Request) {
 		ps[i] = productView{p, price}
 	}
 
+	if os.Getenv("CONVERT_CURRENCIES") == "Active" {
+		for i := 0 ; i < 100; i++ { 
+			for _, p := range products {
```

Comment:
This is starting to approach to many nesting layers to be easily understood.  I think this is OK, but in general you should think about decomposing into separate functions just to keep things more readable, not just DRY.

Reply:
Yeah I agree - I did a refactor that hopefully makes it easier to understand! I pulled the conversion piece into its own function